### ISSUE TITLE
[Feature Request]: Add a basic doc explaining Beam's security model

### ISSUE THREAD

**Issue by aaltay** [apache/beam#30911]
### What would you like to happen?

Explain a few concepts like:
- Beam allows users to run arbitrary code. And Beam does not have different code privilege levels. (Because of that some code execution vulnerabilities will be normal within Beam's model.)
- Beam work with different runners. Runners security models will apply in the execution environment (e.g. cluster models might allow resource access across running jobs.) (Because of that runner related security issues will be best addressed with the specific runners.)
- ...

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [X] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by aaltay** [comment]
It was a discussion on the Beam's private list (among the PMC) as a follow up to security report.

**Comment by rakeshcusat** [comment]
@aaltay Was there any consensus where this note will exist? I am assuming that keeping in our beam website, and individual SDK documentation would make more sense. What do you think?

**Comment by aaltay** [comment]
No, there were no details or consensus on the solution. IMO, keeping this in the beam website (somewhere under https://beam.apache.org/documentation/) would make sense.

If you are interested, please take a first stab. We will appreciate it very much. And it does not need to be perfect, something simple would be a great improvement.
